miner/payload_building.go

```diff
@@ -47,6 +47,8 @@ type BuildPayloadArgs struct {
 	NoTxPool     bool                 // Optimism addition: option to disable tx pool contents from being included
 	Transactions []*types.Transaction // Optimism addition: txs forced into the block via engine API
 	GasLimit     *uint64              // Optimism addition: override gas limit of the block to build
+
+	ExtraData []byte // Flashbots addition: extra data to include in the block
```

Comment:
You should not modify this data structure as it comes from the execution api spec https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#payloadattributesv3 and also we do not use this pattern on L1 https://github.com/flashbots/builder/blob/main/miner/payload_building.go